File: Easy/is-anagram.py
#
# Problem: Is Anagram
# Difficulty: Easy
# URL: https://neetcode.io/problems/is-anagram
# Date: 2026-01-06
#

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        hash1 = {}
        hash2 = {}

        for i in range (len(s)):
            hash1[s[i]] = hash1.get(s[i], 0) + 1
            hash2[t[i]] = hash2.get(t[i], 0) + 1
        return hash1 == hash2

File: Unknown/is-anagram.c
/*
 * Problem: Is Anagram
 * URL: https://neetcode.io/problems/is-anagram
 * Date: 2026-03-07
 */

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # two hashmaps, key=character : value=occurences
        # return true if hash1=hash2, else false
        # edge cases: either a or b is empty or a and b are not same size

        if len(s) != len(t):
            return False


        hash1={}
